src/ai/backend/agent/runtime.py

```diff
@@ -30,27 +53,59 @@ def __init__(
         agent_public_key: Optional[PublicKey],
     ) -> None:
         self.local_config = local_config
+        self.agents = {}
         self.kernel_registry = KernelRegistry()
         self.etcd = etcd
-        self.etcd_view = AgentEtcdClientView(etcd, self.local_config)
+        self.etcd_views = {}
+        self.metadata_server = None
 
+        agent_configs = self.local_config.get_agent_configs()
+        self._default_agent_id = AgentId(agent_configs[0].agent.id)
         self._stop_signal = signal.SIGTERM
 
         self.stats_monitor = stats_monitor
         self.error_monitor = error_monitor
         self.agent_public_key = agent_public_key
 
     async def __ainit__(self) -> None:
-        self.agent = await self._create_agent(self.etcd_view, self.local_config)
-
-    async def __aexit__(self, *exc_info) -> None:
-        await self.agent.shutdown(self._stop_signal)
+        tasks = []
+        async with asyncio.TaskGroup() as tg:
+            for agent_config in self.local_config.get_agent_configs():
+                agent_id = AgentId(agent_config.agent.id)
+                etcd_view = AgentEtcdClientView(self.etcd, agent_config)
 
-    def get_agent(self) -> AbstractAgent:
-        return self.agent
+                self.etcd_views[agent_id] = etcd_view
+                tasks.append(tg.create_task(self._create_agent(etcd_view, agent_config)))
+        self.agents = {(agent := task.result()).id: agent for task in tasks}
 
-    def get_etcd(self) -> AgentEtcdClientView:
-        return self.etcd_view
+    async def __aexit__(self, *exc_info) -> None:
+        for agent in self.agents.values():
+            await agent.shutdown(self._stop_signal)
+        if self.metadata_server is not None:
+            await self.metadata_server.cleanup()
+
+    def get_agents(self) -> list[AbstractAgent]:
+        return list(self.agents.values())
+
+    def get_agent(self, agent_id: Optional[AgentId]) -> AbstractAgent:
+        if agent_id is None:
+            agent_id = self._default_agent_id
```

Comment:
If we handle it with `_default_agent` instead of `_default_agent_id`, I think we can early return and not have to worry about whether there will be an impact on the code afterwards.